internal/topic/topicwriterinternal/writer_reconnector.go

```diff
@@ -112,6 +112,22 @@ func NewWriterReconnectorConfig(options ...PublicWriterOption) WriterReconnector
 		WithProducerID(uuid.NewString())(&cfg)
 	}
 
+	if cfg.Connect == nil {
```

Comment:
The confgin should not store state and should not change self content - it is un-intuitive and difficult to understande when research the code.